lib/rules/no-dupe-disjunctions.ts

```diff
@@ -1104,6 +1145,103 @@ export default createRule("no-dupe-disjunctions", {
                 }
             }
 
+            /** Prints the given character. */
+            function printChar(char: number): string {
+                if (inRange(allowedRanges, char)) {
+                    return String.fromCodePoint(char)
+                }
+
+                if (char === 0) return "\\0"
+                if (char <= 0xff)
+                    return `\\x${char.toString(16).padStart(2, "0")}`
+                if (char <= 0xffff)
+                    return `\\u${char.toString(16).padStart(4, "0")}`
+
+                return `\\u{${char.toString(16)}}`
+            }
+
+            /** Returns suggestions for fixing the given report */
+            function getSuggestions(
+                result: Result,
+            ): Rule.SuggestionReportDescriptor[] {
+                const alternative =
+                    result.type === "NestedSubset" ||
+                    result.type === "PrefixNestedSubset"
+                        ? result.nested
+                        : result.alternative
+
+                const containsCapturingGroup = hasSomeDescendant(
+                    alternative,
+                    (d) => d.type === "CapturingGroup",
+                )
+                if (containsCapturingGroup) {
+                    // we can't just remove a capturing group
+                    return []
+                }
+
+                if (result.type === "Overlap" || result.type === "Superset") {
+                    // the types of results cannot be trivially fixed by
+                    // removing an alternative.
+                    return []
+                }
```

Comment:
I prefer to move this branch to the beginning of the function, as I think the type checking on line 1168 will be more stringent.
What do you think?

Reply:
Good idea!